src/MongoDB.Bson/IO/JsonWriter.cs

```diff
@@ -886,6 +886,10 @@ private string GuidToString(BsonBinarySubType subType, byte[] bytes, GuidReprese
                 var message = string.Format("Length of binary subtype {0} must be 16, not {1}.", subType, bytes.Length);
                 throw new ArgumentException(message);
             }
+            if (guidRepresentation == GuidRepresentation.Unspecified && subType == BsonBinarySubType.UuidStandard)
+            {
+                guidRepresentation = GuidRepresentation.Standard;
+            }
```

Comment:
I wondered why you had swapped the order of the tests from my original suggestion... I see your point though.

I suppose a more logical refactoring of the various tests that avoids duplication and organizes them a bit better would be:

    if (subType == BsonBinarySubType.UuidLegacy)
    {
        if (guidRepresentation == GuidRepresentation.Standard)
        {
            throw new ArgumentException("GuidRepresentation for binary subtype UuidLegacy must not be Standard.");
        }
    }
    if (subType == BsonBinarySubType.UuidStandard)
    {
        if (guidRepresentation == GuidRepresentation.Unspecified)
        {
            guidRepresentation = GuidRepresentation.Standard;
        }
        if (guidRepresentation != GuidRepresentation.Standard)
        {
            var message = string.Format("GuidRepresentation for binary subtype UuidStandard must be Standard, not {0}.", guidRepresentation);
            throw new ArgumentException(message);
        }
    }

We can go with your version if you want. I just didn't see why you had swapped the clauses.

Reply:
I applied the latest suggestion, it looks better